PINCache/PINDiskCache.m

```diff
@@ -457,6 +457,22 @@ - (void)lockFileAccessWhileExecutingBlock:(void(^)(PINDiskCache *diskCache))bloc
     });
 }
 
+- (void)containsObjectForKey:(NSString *)key block:(PINDiskCacheContainsBlock)block
+{
+    __weak PINDiskCache *weakSelf = self;
+    
+    dispatch_async(_asyncQueue, ^{
+        PINDiskCache *strongSelf = weakSelf;
+        BOOL containsObject = [strongSelf containsObjectForKey:key];
+        
+        if (block) {
+            [strongSelf lock];
```

Comment:
Maybe I'm overlooking something, but why do we need to lock on reporting the result? Should the lock instead of around the `-containsObjectForKey:` call?